core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -319,61 +359,96 @@ class LogTest {
 
     log.truncateFullyAndStartAt(29)
     assertEquals(1, log.logSegments.size)
-    assertEquals(None, log.latestProducerSnapshotOffset)
+    assertEquals(Some(29), log.latestProducerSnapshotOffset)
```

Comment:
perhaps we should also assert that `log.oldestProducerSnapshotOffset` is also the same value?